### DISCUSSION TITLE
OCR for quick entries

### DISCUSSION THREAD

**Discussion by ThisIsQasim** [akhilrex/hammond#136] [Ideas]
I am thinking it would be super useful if Hammond could use OCR to pick up the details of a fill up from a quick entry and autocomplete those fields when that quick entry is referred in the fill up.

**Comment by Greal-dev** [comment]
If anyone is looking for a solution to this, I built an open-source tool called [Car Carer](https://github.com/Greal-dev/car-carer) that does exactly this — upload a PDF or photo of an invoice/receipt, and AI (Gemini Flash 2.0) automatically extracts all the structured data (date, mileage, garage, parts, costs, vehicle info).

It also has an intelligent chat assistant (Claude) to query your maintenance history in natural language.

It's a standalone self-hosted app (FastAPI + SQLite, Docker ready), not a Hammond plugin, but could be useful as a complementary tool. MIT licensed.

https://github.com/Greal-dev/car-carer
